### ISSUE TITLE
Building PHP 7 with external gd failed

### ISSUE THREAD

**Issue by spajak** [libgd/libgd#781]
Compilation of PHP 7 fails when --with-external-gd (with --enable-gd) flag is active.
GD_FLIP_X constants are somehow undeclared.
GD version 2.3.1 worked fine. The problem is with 2.3.3.

```
/bin/bash /home/USER/build/php-7.4.24/libtool --silent --preserve-dup-deps --mode=compile cc  -Iext/gd/ -I/home/USER/build/php-7.4.24/ext/gd/ -DPHP_ATOM_INC -I/home/USER/build/php-7.4.24/include -I/home/USER/build/php-7.4.24/main -I/home/USER/build/php-7.4.24 -I/home/USER/build/php-7.4.24/ext/date/lib -I/usr/include/libxml2 -I/usr/local/include -I/usr/include/x86_64-linux-gnu -I/usr/local/include/freetype2 -I/usr/local/include/libpng16 -I/home/USER/build/php-7.4.24/ext/mbstring/libmbfl -I/home/USER/build/php-7.4.24/ext/mbstring/libmbfl/mbfl -I/home/USER/build/php-7.4.24/TSRM -I/home/USER/build/php-7.4.24/Zend  -D_REENTRANT -pthread  -I/usr/include -O2 -Wno-error -fvisibility=hidden -pthread -Wall -Wno-strict-aliasing -DZTS -DZEND_SIGNALS   -c /home/USER/build/php-7.4.24/ext/gd/gd.c -o ext/gd/gd.lo
In file included from /home/USER/build/php-7.4.24/main/php.h:441,
                 from /home/USER/build/php-7.4.24/ext/gd/gd.c:30:
/home/USER/build/php-7.4.24/ext/gd/gd.c: In function ‘zm_startup_gd’:
/home/USER/build/php-7.4.24/ext/gd/gd.c:1140:48: error: ‘GD_FLIP_HORINZONTAL’ undeclared (first use in this function)
 1140 |  REGISTER_LONG_CONSTANT("IMG_FLIP_HORIZONTAL", GD_FLIP_HORINZONTAL, CONST_CS | CONST_PERSISTENT);
      |                                                ^~~~~~~~~~~~~~~~~~~
/home/USER/build/php-7.4.24/Zend/zend_constants.h:53:105: note: in definition of macro ‘REGISTER_LONG_CONSTANT’
   53 | #define REGISTER_LONG_CONSTANT(name, lval, flags)  zend_register_long_constant((name), sizeof(name)-1, (lval), (flags), module_number)
      |                                                                                                         ^~~~
/home/USER/build/php-7.4.24/ext/gd/gd.c:1140:48: note: each undeclared identifier is reported only once for each function it appears in
 1140 |  REGISTER_LONG_CONSTANT("IMG_FLIP_HORIZONTAL", GD_FLIP_HORINZONTAL, CONST_CS | CONST_PERSISTENT);
      |                                                ^~~~~~~~~~~~~~~~~~~
/home/USER/build/php-7.4.24/Zend/zend_constants.h:53:105: note: in definition of macro ‘REGISTER_LONG_CONSTANT’
   53 | #define REGISTER_LONG_CONSTANT(name, lval, flags)  zend_register_long_constant((name), sizeof(name)-1, (lval), (flags), module_number)
      |                                                                                                         ^~~~
/home/USER/build/php-7.4.24/ext/gd/gd.c:1141:46: error: ‘GD_FLIP_VERTICAL’ undeclared (first use in this function)
 1141 |  REGISTER_LONG_CONSTANT("IMG_FLIP_VERTICAL", GD_FLIP_VERTICAL, CONST_CS | CONST_PERSISTENT);
      |                                              ^~~~~~~~~~~~~~~~
/home/USER/build/php-7.4.24/Zend/zend_constants.h:53:105: note: in definition of macro ‘REGISTER_LONG_CONSTANT’
   53 | #define REGISTER_LONG_CONSTANT(name, lval, flags)  zend_register_long_constant((name), sizeof(name)-1, (lval), (flags), module_number)
      |                                                                                                         ^~~~
/home/USER/build/php-7.4.24/ext/gd/gd.c:1142:42: error: ‘GD_FLIP_BOTH’ undeclared (first use in this function)
 1142 |  REGISTER_LONG_CONSTANT("IMG_FLIP_BOTH", GD_FLIP_BOTH, CONST_CS | CONST_PERSISTENT);
      |                                          ^~~~~~~~~~~~
/home/USER/build/php-7.4.24/Zend/zend_constants.h:53:105: note: in definition of macro ‘REGISTER_LONG_CONSTANT’
   53 | #define REGISTER_LONG_CONSTANT(name, lval, flags)  zend_register_long_constant((name), sizeof(name)-1, (lval), (flags), module_number)
      |                                                                                                         ^~~~
/home/USER/build/php-7.4.24/ext/gd/gd.c: In function ‘zif_imageflip’:
/home/USER/build/php-7.4.24/ext/gd/gd.c:4699:8: error: ‘GD_FLIP_VERTICAL’ undeclared (first use in this function)
 4699 |   case GD_FLIP_VERTICAL:
      |        ^~~~~~~~~~~~~~~~
/home/USER/build/php-7.4.24/ext/gd/gd.c:4703:8: error: ‘GD_FLIP_HORINZONTAL’ undeclared (first use in this function)
 4703 |   case GD_FLIP_HORINZONTAL:
      |        ^~~~~~~~~~~~~~~~~~~
/home/USER/build/php-7.4.24/ext/gd/gd.c:4707:8: error: ‘GD_FLIP_BOTH’ undeclared (first use in this function)
 4707 |   case GD_FLIP_BOTH:
      |        ^~~~~~~~~~~~
make: *** [Makefile:931: ext/gd/gd.lo] Error 1
```

Linux 5.10.0-8-amd64 #1 SMP Debian 5.10.46-4 (2021-08-03) x86_64 GNU/Linux

**Comment by cmb69** [comment]
Duplicate of #777.
